shell/browser/api/atom_api_url_request.cc

```diff
@@ -524,8 +536,13 @@ void URLRequest::DoWrite() {
 }
 
 void URLRequest::StartWriting() {
-  if (!last_chunk_written_ || size_callback_.is_null())
+  if (!last_chunk_written_ || size_callback_.is_null() || !producer_)
```

Comment:
Do we need to emit some event here in the case of `!producer_`?

Reply:
Personally don't think so, `StartWriting` is now eager and either ns or urlrequest calls it, and if we call it producer doesn't exist we'll wait for next ns callback.